lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -131,11 +131,15 @@ def authorize_ingress(new_rules, existing_rules=[])
     to_create.reject(&:nil?).each do |rule|
       if rule.key? 'security_group'
         source_group_name = rule['security_group']
-        group_response = ec2.describe_security_groups(filters: [
-          {name: 'group-name', values: [source_group_name]},
-        ])
+        filters = [ {name: 'group-name', values: [source_group_name]} ]
+        if vpc_only_account?
+          response = ec2.describe_security_groups(group_ids: [@property_hash[:id]])
+          vpc_id = response.data.security_groups.first.vpc_id
+          filters.push( {name: 'vpc-id', values: [vpc_id]} )
+        end
+        group_response = ec2.describe_security_groups(filters: filters)
         fail("No groups found called #{source_group_name}") if group_response.data.security_groups.count == 0
-        source_group_id = group_response.data.security_groups.first.group_id
+        source_group_id = group_response.data.security_groups.find { |g| g.group_name == source_group_name }.group_id
```

Comment:
Yes, I've confirmed that the only real issue here is that we weren't previously filtering on `vpc_id`.  Calling `find` shouldn't actually be necessary as was first assumed.